### ISSUE TITLE
Pre-workshop survey questions?

### ISSUE THREAD

**Issue by weiji14** [GenericMappingTools/agu24workshop#20]
@Esteban82 mentioned at https://github.com/GenericMappingTools/agu24workshop/issues/13#issuecomment-2497978550 that we should send out a survey to the participants (now confirmed to be 32 people). Here's a rough list I'm thinking of:

- Operating system (Windows, macOS, Linux, other)
- Familiarity with GMT (beginner, intermediate, expert)
- Familiarity with PyGMT (beginner, intermediate, expert)
- Career level (undergrad, Masters, PhD, postdoc, etc)

Are there other questions people would like to know? E.g. we could have a free-form field about their study field (e.g. Seismology, Planetary Science, etc).

@maxrjones, I think we both have the attendee registration list, so maybe we both can coordinate on this? Let's try to send this out by Wednesday afternoon EST perhaps.

**Comment by maxrjones** [comment]
Happy to help with this. I can draft an email tomorrow.

Would we revise any material based on the responses to those questions? If not, I'm wondering if it might be nicer to use those as icebreaker questions at the start of the workshop, to get people in the habit of engaging and allow everyone to see who's in a similar boat to them.

**Comment by weiji14** [comment]
I doubt we'll have time to revise any workshop material (there's still some final polish needed), so using those questions as icebreakers is a great idea actually, I'll include that in the 10min introduction :smile:

That said, we should still send out an email with some general info, e.g. hardware requirements (bring a laptop), workshop venue, link to the repository/website, installation instructions, etc. We should still try to send this out by Wednesday, but not so urgent as before if we're not expecting any feedback response (from a survey).

**Comment by yvonnefroehlich** [comment]
> Would we revise any material based on the responses to those questions?

I agrere with @weiji14 revising the workshop material in a major way is not realistic.

> [...] I'm wondering if it might be nicer to use those as icebreaker questions at the start of the workshop, to get people in the habit of engaging and allow everyone to see who's in a similar boat to them.

I like this idea 🙂!

> Are there other questions people would like to know? E.g. we could have a free-form field about their study field (e.g. Seismology, Planetary Science, etc).

Hm. Maybe it would also be interesting to know which tools people are currently using to create visualizations of spatial data, especially maps. `Basemap` isn't supported anymore, so maybe people think about what to use now? And do not simply move to `cartopy` but consider pther tools like GeoPandas and PyGMT. Or if they simply moved from GMT to PyGMT.

**Comment by Esteban82** [comment]
Another questions that I would include are:

1. Experience with bash. It is related to experience with GMT.
2. If they manage to install GMT and PyGMT or if they would need help.
3. Among the instructors we speak several languages. Perhaps we can ask if they speak another language to see if they can consult with a particular instructor. Or maybe we, when we introduce ourselves, say in what language we can be asked. 
4. I would ask about their study field. Later, however, we ask them to tell us about it when they present themselves.

**Comment by maxrjones** [comment]
> That said, we should still send out an email with some general info, e.g. hardware requirements (bring a laptop), workshop venue, link to the repository/website, installation instructions, etc. We should still try to send this out by Wednesday, but not so urgent as before if we're not expecting any feedback response (from a survey).

Sounds good, I'll draft an email as soon as we get https://github.com/GenericMappingTools/agu24workshop/pull/12 merged.

**Comment by weiji14** [comment]
Cool, thanks everyone for the ideas! I'm taking notes and will make sure to use these in the introduction!

> > Hm. Maybe it would also be interesting to know which tools people are currently using to create visualizations of spatial data, especially maps. `Basemap` isn't supported anymore, so maybe people think about what to use now? And do not simply move to `cartopy` but consider pther tools like GeoPandas and PyGMT. Or if they simply moved from GMT to PyGMT.
>
> > 1. Experience with bash. It is related to experience with GMT.

These are the sorts of questions I would like to ask in a GMT user survey (GDAL recently put out their 2024 survey results at https://gdal.org/en/stable/community/user_survey_2024.html). Given that Paul's memorial session will be taking place too, maybe this is a good chance to create a wider GMT survey and advertise it during the AGU conference? (Does anyone want to take the lead on this)?

> 3. Among the instructors we speak several languages. Perhaps we can ask if they speak another language to see if they can consult with a particular instructor. Or maybe we, when we introduce ourselves, say in what language we can be asked.
> 
> 4. I would ask about their study field. Later, however, we ask them to tell us about it when they present themselves.

On point 3, yes, we've got a pretty international group (both instructors and participants), so I'll make sure to give you all the chance to do a brief introduction at the start as we go through the workshop outline :smile: On point 4, I don't think we have time to get everyone to talk individually at the start, but I could ask general questions (e.g. how far they flew to get to AGU, how many languages they speak, etc). Hopefully we'll get to know them better during lunch or in the final exercises/mini-project.

**Comment by maxrjones** [comment]
@weiji14 here's a draft email - https://docs.google.com/document/d/1xUr14ZpTmEG8r5UN2h8oyHVlT6GTHXri3eL4IEq89M4/edit?usp=sharing

**Comment by weiji14** [comment]
Thanks @maxrjones! The installation instructions are up at https://www.generic-mapping-tools.org/agu24workshop/intro.html#running-the-notebooks-locally now. Feel free to send the email when you're free today or tomorrow. @Esteban82 did mention that Windows users should something like [`Git for Windows`](https://gitforwindows.org/) installed to have a bash prompt for the animation tutorial, which we could add somewhere later (edit: see #25).

**Comment by jhtong33** [comment]
> These are the sorts of questions I would like to ask in a GMT user survey (GDAL recently put out their 2024 survey results at https://gdal.org/en/stable/community/user_survey_2024.html). Given that Paul's memorial session will be taking place too, maybe this is a good chance to create a wider GMT survey and advertise it during the AGU conference? (Does anyone want to take the lead on this)?

@weiji14  Do we still need a survey now? I can quickly create a version using Google Forms right away.

**Comment by weiji14** [comment]
> @weiji14 Do we still need a survey now? I can quickly create a version using Google Forms right away.

Not for the workshop, but maybe a more general audience. If you want to start, please go ahead, and we can advertise it on the GMT forum and mention it somehow next Monday during AGU.

**Comment by maxrjones** [comment]
> > @weiji14 Do we still need a survey now? I can quickly create a version using Google Forms right away.
> 
> Not for the workshop, but maybe a more general audience. If you want to start, please go ahead, and we can advertise it on the GMT forum and mention it somehow next Monday during AGU.

For some background, I suggested a GMT survey similar to GDAL's when I started as a post-doc in 2022. GMT core devs suggested that it may be counterproductive. In particular, there were concerns that we would not have the capacity to address the users requests and seeking feedback without acting on it would erode trust. They suggested that it could be appropriate before submitting a funding proposal or if we were to have general maintenance funding (like GDAL has), but not if we already had committed to certain deliverables through a proposal, or capacity to shift development direction due to limited time.
